Detections/ASimAuthentication/imAuthSigninsMultipleCountries.yaml

```diff
@@ -19,15 +19,16 @@ tags:
   - Schema: ASIMAuthentication
     SchemaVersion: 0.1.0
 query: |
-  let timeframe = ago(3h);
+  let searchTimeframe = ago(3h);
   let threshold = 2;
+  let ipWhitelist = dynamic([]);
   imAuthentication
-  | where TimeGenerated > timeframe
-  | where EventType=='Logon' and EventResult=='Success'
+  | where TimeGenerated > searchTimeframe
+  | where not(SrcIpAddr has_any (ipWhitelist))
+  | where EventType == 'Logon' and EventResult == 'Success'
   | where isnotempty(SrcGeoCountry)
-  | summarize StartTime = min(TimeGenerated), EndTime = max(TimeGenerated), Vendors=make_set(EventVendor), Products=make_set(EventProduct)
-    , NumOfCountries = dcount(SrcGeoCountry)
-    by TargetUserId, TargetUsername, TargetUserType
+  | summarize StartTime = min(TimeGenerated), EndTime = max(TimeGenerated), Vendors=make_set(EventVendor), Products=make_set(EventProduct), IPs = make_set(SrcIpAddr), Countries = make_set(SrcGeoCountry), NumOfCountries = dcount(SrcGeoCountry)
```

Comment:
If we have already summarized a set of Countries, we do not need to have distinct count. This can be calculated using array_length function.

Reply:
I just copied over the library query for this, I am not sure how to fix this in KQL.

Reply:
You are using NumOfCountries = dcount(SrcGeoCountry) in the line 30.
Instead of that, in line 31 you can have 
| extend NumOfCountries = array_length(set_SrcGeoCountry)